### DISCUSSION TITLE
The value of the transmit power get by the ot-ctl txpower command is different from the value set by ot-ctl txpower <txpower>. Why?

### DISCUSSION THREAD

**Discussion by jinpeng1989** [openthread/openthread#9104] [Q&A] [answered]
For example, if it's set to 9dBm, it actually reads 8dBm. Could you please help to make it clear? Is there any correspondence between the two values? Could you please indicate it in the code?

![image](https://github.com/openthread/openthread/assets/97089879/ff98b3d4-09d6-4136-8da3-1b138d6b68a4)

Set Value | Get Value
-- | --
9 | 8
8 | 7
7 | 7
6 | 5
5 | 5
4 | 3
3 | 2
2 | 2
1 | 0
0 | -1
-1 | -1
-2 | -2
-3 | -2
-4 | -4

**Comment by jwhui** [comment]
The `txpower` CLI command directly calls the platform APIs:

https://github.com/openthread/openthread/blob/c187e4d049a980cf2fcf7b3f20d1c2e712b6f5af/src/cli/cli.cpp#L6956-L6974

I would reach out to your platform vendor to inquire about the radio transmit power APIs.

**Reply by jinpeng1989** [reply]
Thanks, here's the RCP SoC and SDK information.
RCP SoC:  [Silicon Labs EFR32MG21](https://www.silabs.com/wireless/zigbee/efr32mg21-series-2-socs)
SDK: Gecko SDK (GSDK) Version [v4.2.3.0](https://github.com/SiliconLabs/gecko_sdk/releases/tag/v4.2.3)

**Reply by jwhui** [reply]
As mentioned, these CLI commands map to platform-specific APIs.

For EFR32, it is:
- https://github.com/openthread/ot-efr32/blob/c67a82016e2bad9e11054a0250808d64c7d7b334/src/src/radio.c#L1865
- https://github.com/openthread/ot-efr32/blob/c67a82016e2bad9e11054a0250808d64c7d7b334/src/src/radio.c#L1846

**Reply by silabs-grace** [reply, marked as answer]
The platform specific API documentation can be found here: https://docs.silabs.com/rail/2.14/rail-api/pa#rail-set-tx-power. 

This most likely is related to the RAIL_SetTxPower() function. Depending on the RAIL_ConfigTxPower configuration not all power levels are achievable, this API will get as close as possible to the desired power without exceeding it. You can find more information here: https://www.silabs.com/documents/public/application-notes/an1127-power-amplifier-power-conversion-functions.pdf
